hydrolib/core/io/storagenode/models.py

```diff
@@ -0,0 +1,225 @@
+            "Storage area from streetLevel upwards if useStreetStorage = true [m2].",
+            alias="streetStorageArea",
+        )
+        storagetype: Optional[str] = Field(
+            "Possible values: "
+            + "reservoir: Above streetLevel the storage area of this node is also taken into account. "
+            + "closed: Above streetLevel this storage node has no storage area.",
+            alias="storageType",
+        )
+        numlevels: Optional[str] = Field(
+            "Number of levels in storage area table.", alias="numLevels"
+        )
+        levels: Optional[str] = Field(
+            "Levels in storage area table [m].", alias="levels"
+        )
+        storagearea: Optional[str] = Field(
+            "Areas in storage area table [m2].", alias="storageArea"
+        )
+        interpolate: Optional[str] = Field(
+            "Interpolation type for storage area table either linear or block.",
+            alias="interpolate",
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["StorageNode"] = "StorageNode"
+    id: str = Field(alias="id")
+    name: str = Field(alias="name")
+    manholeid: Optional[str] = Field(alias="manholeId")
+
+    nodetype: Optional[NodeType] = Field(NodeType.unspecified, alias="nodeType")
+    nodeid: str = Field(alias="nodeId")
+    usetable: bool = Field(alias="useTable")
+
+    # useTable is True
+    bedlevel: Optional[float] = Field(alias="bedLevel")
+    area: Optional[float] = Field(alias="area")
+    streetlevel: Optional[float] = Field(alias="streetLevel")
+    streetstoragearea: Optional[float] = Field(alias="streetStorageArea")
+    storagetype: Optional[StorageType] = Field(
+        StorageType.reservoir, alias="storageType"
+    )
+
+    # useTable is False
+    numlevels: Optional[int] = Field(alias="numLevels")
+    levels: Optional[List[float]] = Field(alias="levels")
+    storagearea: Optional[List[float]] = Field(alias="storageArea")
+    interpolate: Optional[Interpolation] = Field(
+        Interpolation.linear, alias="interpolate"
+    )
+
+    _interpolation_validator = get_enum_validator("interpolate", enum=Interpolation)
+    _nodetype_validator = get_enum_validator("nodetype", enum=NodeType)
+    _storagetype_validator = get_enum_validator("storagetype", enum=StorageType)
+    _split_to_list = get_split_string_on_delimiter_validator(
+        "levels",
+        "storagearea",
+        delimiter=" ",
+    )
+
+    def _get_identifier(self, data: dict) -> Optional[str]:
+        return data.get("id") or data.get("name")
+
+    @root_validator(skip_on_failure=True)
+    @classmethod
+    def _validate_usetable(cls, values):
+
+        if values["usetable"]:
+            cls._validate_required_usetable_fields(
+                "numlevels", "levels", "storagearea", values=values, usetable=True
+            )
+
+            cls._validate_list_length(
+                "levels", "storagearea", length_field_name="numlevels", values=values
```

Comment:
Can you try and replace this with the existing utility, for example:
```
    _check_list_length = make_list_length_root_validator(
        "levels",
        "storagearea",
        length_name="numlevels",
        list_required_with_length=True,
    )
```
(last argument list_required_with_length comes from #179).

Reply:
Thanks, I will do that once it is in the master :)

Reply:
Done